homeassistant/components/tankerkoenig/sensor.py

```diff
@@ -0,0 +1,109 @@
+"""Tankerkoenig sensor integration."""
+
+import logging
+
+from homeassistant.const import (
+    ATTR_ATTRIBUTION,
+    ATTR_LATITUDE,
+    ATTR_LONGITUDE,
+    STATE_CLOSED,
+    STATE_OPEN,
+    STATE_UNKNOWN,
+)
+from homeassistant.helpers.entity import Entity
+
+from .const import DOMAIN, NAME
+
+_LOGGER = logging.getLogger(__name__)
+
+ATTR_BRAND = "brand"
+ATTR_CITY = "city"
+ATTR_FUEL_TYPE = "fuel_type"
+ATTR_HOUSE_NUMBER = "house_number"
+ATTR_IS_OPEN = "is_open"
+ATTR_POSTCODE = "postcode"
+ATTR_STATION_NAME = "station_name"
+ATTR_STREET = "street"
+ATTRIBUTION = "Data provided by https://creativecommons.tankerkoenig.de"
+
+ICON = "mdi:gas-station"
+
+
+async def async_setup_platform(hass, config, async_add_entities, discovery_info=None):
+    """Set up the tankerkoenig sensors."""
+    tankerkoenig = hass.data[DOMAIN]
```

Comment:
Please add a guard clause here that checks if discovery_info is None and return if so. We need to guard against the user adding a platform config section by mistake which would load the platform twice.